pypeit/spec2dobj.py

```diff
@@ -415,22 +416,59 @@ def write_to_fits(self, outfile, pri_hdr=None, update_det=None, overwrite=True):
-                Detector to be updated
+                Detector to be updated.  If given, all of the other
+                detectors remain as they were in the file (if the file exists)
+            slitspatnum (:obj:`str` or :obj:`list`, optional):
+              Restricted set of slits for reduction
+              If provided, do not clobber the existing file but only update
+              the indicated slits.  Useful for re-running on a subset of slits
             overwrite (bool, optional):
 
         """
         if os.path.isfile(outfile):
+            # Clobber?
             if not overwrite:
                 msgs.warn("File {} exits.  Use -o to overwrite.".format(outfile))
                 return
+            # Load up the original
+            _allspecobj = AllSpec2DObj.from_fits(outfile)
+            # Replace the newly reduced detector?
             if update_det is not None:
-                # Load up and replace
-                _allspecobj = AllSpec2DObj.from_fits(outfile)
                 for det in _allspecobj.detectors:
                     if det in np.atleast_1d(update_det):
                         continue
                     else:
                         self[det] = _allspecobj[det]
+            elif slitspatnum is not None: # Update specific slits!
+                
+                # Grab modified detectors and slits
+                dets, spat_ids = slittrace.parse_slitspatnum(slitspatnum)
+
+                # Loop on detectors to be fussed with
+                for det in _allspecobj.detectors:
+                    if det in dets:
+                        # Check version 
+                        if self[det].version != _allspecobj[det].version:
+                            msgs.error("Original spec2D object has a different version.  Too risky to continue.  Rerun both")
+                        # Generate the slit "mask"
+                        slitmask = _allspecobj[det].slits.slit_img(
+                            flexure=_allspecobj[det].sci_spat_flexure)
+                        # Save the new one in a copy
+                        new_Spec2DObj = deepcopy(self[det])
+                        # Replace with the old
+                        self[det] = _allspecobj[int(det)]
+                        # Spat ids    
+                        spats = spat_ids[dets==det]
+                        for spat_id in spats:
+                            # Find pixels to replace
+                            replace_pix = slitmask == spat_id
+                            # Fill em in
+                            for key in new_Spec2DObj.datamodel.keys():
+                                if new_Spec2DObj.datamodel[key]['otype'] == np.ndarray and (
+                                    new_Spec2DObj[key].shape == slitmask.shape):
+                                    self[det][key][replace_pix] = new_Spec2DObj[key][replace_pix]
+                    else:
+                        self[det] = _allspecobj[det]
```

Comment:
Much of this may need to be updated given the changes from the `mosaic` branch.  Let me know if you need help.